src/components/App.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <div id="app">
+    <template v-if="$store.state.user">
+      <h2>You are logged in!</h2>
+      <div class="logout">
+        <button @click="logout">Logout</button>
+      </div>
+      <router-view></router-view>
+    </template>
+    <template v-else>
+      <div class="login">
+        <Auth />
+      </div>
+    </template>
+  </div>
+</template>
+
+<script>
+import Auth from './Auth';
+
+export default {
+  name: 'app',
+  components: {
+    Auth,
+  },
+  methods: {
+    logout() {
+      this.$store.dispatch('LOGOUT');
+    },
+  },
+  mounted() {
+    this.$store.dispatch('AUTH_CHECK');
```

Comment:
This is checking the store to see what the authentication status of the user is early in the page load, so that the v-of statement above will be able to resolve to true or false